lib/csv.rb

```diff
@@ -905,20 +905,49 @@ def initialize(message, line_number)
     },
     all:       [:date_time, :numeric],
   }
+  Ractor.make_shareable(DefaultConverters) if defined?(Ractor)
+  Converters = DefaultConverters.dup
```

Comment:
Ah, this keeps backward compatibility.
But Ractor users can't use converters added by users.

Reply:
I agree that if a user defines a converter in the main Ractor, then the converter can't automatically be used from non-main-Ractors. This change would allow users to define converters within one Ractor and use them just within the same Ractor. This can be done by getting the Ractor-local converters hash with get_converters, and then adding to it.

Without the change, converters are completely inaccessible from Ractors, so I think having Ractor-local versions is a step forward. Since it does not currently seem possible to have shared hashes among Ractors, I am not sure how the ideal situation could be achieved.

Reply:
I decided API for `Converters` and `HeaderConverters`:

* If a Ractor user doesn't use any built-in converters, the user can use `CSV`. (committed)
* If a Ractor user wants to use any built-in converters in non-main-Ractors, the user must call `Ractor.make_sharable(CSV::Converters)` and `Ractor.make_sharedable(CSV::HeaderConverters)` after the user adds additional converters and before non-main-Ractors are created.
  * This should be documented. (Could you work on it in this pull request?)